csharp/src/Apache.Arrow.Flight.Sql/Client/FlightSqlClient.cs

```diff
@@ -0,0 +1,1211 @@
+        {
+            var command = new CommandGetSqlInfo();
+            var descriptor = FlightDescriptor.CreateCommandDescriptor(command.PackAndSerialize());
+            var schemaResultCall = _client.GetSchema(descriptor, options.Headers);
+            var schemaResult = await schemaResultCall.ResponseAsync.ConfigureAwait(false);
+
+            return schemaResult;
+        }
+        catch (RpcException ex)
+        {
+            throw new InvalidOperationException("Failed to get SQL info schema", ex);
+        }
+    }
+
+    /// <summary>
+    /// Explicitly cancel a FlightInfo.
+    /// </summary>
+    /// <param name="options">RPC-layer hints for this call.</param>
+    /// <param name="request">The CancelFlightInfoRequest.</param>
+    /// <returns>A Task representing the asynchronous operation. The task result contains the CancelFlightInfoResult describing the canceled result.</returns>
+    public async Task<CancelFlightInfoResult> CancelFlightInfoAsync(FlightCallOptions options,
+        CancelFlightInfoRequest request)
+    {
+        if (options == null) throw new ArgumentNullException(nameof(options));
+        if (request == null) throw new ArgumentNullException(nameof(request));
+
+        try
+        {
+            var action = new FlightAction("CancelFlightInfo", request.PackAndSerialize());
+            var call = _client.DoAction(action, options.Headers);
+            await foreach (var result in call.ResponseStream.ReadAllAsync())
+            {
+                var cancelResult = Any.Parser.ParseFrom(result.Body);
+                if (cancelResult.TryUnpack(out CancelFlightInfoResult cancelFlightInfoResult))
+                {
+                    return cancelFlightInfoResult;
+                }
+            }
+
+            throw new InvalidOperationException("No response received for the CancelFlightInfo request.");
+        }
+        catch (RpcException ex)
+        {
+            throw new InvalidOperationException("Failed to cancel flight info", ex);
+        }
+    }
+
+    /// <summary>
+    /// Explicitly cancel a query.
+    /// </summary>
+    /// <param name="options">RPC-layer hints for this call.</param>
+    /// <param name="info">The FlightInfo of the query to cancel.</param>
+    /// <returns>A Task representing the asynchronous operation.</returns>
+    public async Task<CancelStatus> CancelQueryAsync(FlightCallOptions options, FlightInfo info)
+    {
+        if (options == null)
+        {
+            throw new ArgumentNullException(nameof(options));
+        }
+
+        if (info == null)
+        {
+            throw new ArgumentNullException(nameof(info));
+        }
+
+        try
+        {
+            var cancelRequest = new CancelFlightInfoRequest(info);
+            var action = new FlightAction("CancelFlightInfo", cancelRequest.ToByteString());
+            var call = _client.DoAction(action, options.Headers);
+            await foreach (var result in call.ResponseStream.ReadAllAsync())
```

Comment:
Add `ConfigureAwait(false)` if supported (I've not tried on a foreach)

Reply:
done